bugfix: fixed issue where RTL implementation for foreign keys (e.g. c…

NickvdMeij opened the pull request:
bugfix: fixed issue where RTL implementation for foreign keys (e.g. column name `id_user` for a foreign key) causes problems with LTR tables names (e.g. foreign key to `IdentityProvider` table becomes relation function `getEntityProviders`)

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  |

samdark commented:
That isn't a trivial thing. Is it possible to cover it with test so it doesn't break anymore?

NickvdMeij commented:
Hey @samdark 

Like you asked, I implemented a unit test to set these relations with RTL syntax (id_user instead of user_id) and also the situation where it gave problems (in identity_provider_id foreign key in an organization table). However, I could not get the tests to work locally due to the following error:

`The value of autoloader is specified, but file doesn't exist ''`

even though I did run the composer install and the autoloader file has been generated. Sadly due to time constraints, I was unable to invest more time to sort this out..

Furthermore, I looked at the test results from the previous build, and saw there was one empty (risky) test which I removed (empty function has no use in source code?)

Do you mind looking at the current test implementation (and assuming you got everything running locally) and test is for me? 

Thanks in advance,

Nick van der Meij
MyCademy B.V.

NickvdMeij commented:
@samdark I fixed the syntax errors in the SQL, seems like the tests run as usual in travis CI now. From what i can see, there are still some (minor) problems with the tests, however the relation "getIdentityProviders()" is now generated correctly (was previously "getEntityProviders()" as mentioned in the first description of the PR).

samdark commented:
There are still some breaks: https://travis-ci.org/yiisoft/yii2-gii/jobs/611346074?utm_medium=notification&utm_source=github_status

NickvdMeij commented:
My bad, looking at result of the tests before and after my implementation, I realized the inverseOf's weren't generated at all (also not for the test that was already implemented). Removed the inverseOf's in my tests as well

samdark commented:
OK. Less failures now but still it seems something is broken: https://travis-ci.org/yiisoft/yii2-gii/jobs/611775091

NickvdMeij commented:
@samdark I just commited the (from what I see) last fixes for the tests, my assumption (since I can still not run the tests locally) is that everything should be green now. However, latest commit hasn't been ran by TravisCI. Is there a way to manually trigger this?

NickvdMeij commented:
@samdark last but not least, all tests pass.
Sorry it took a while. Due to the fact that I was unable to test locally there were more commits then there should be. Hopefully this won't be a big problem ;-).

samdark commented:
Not a problem at all.

samdark commented:
Would you please add a line to changelog?

NickvdMeij commented:
@samdark  Added changelog line ;-) Any indication when this will be merged into the master branch? Would love to use this package again instead of my own fork in my php project ;)

samdark commented:
@NickvdMeij soon. Maybe even today.

samdark commented:
Merged. Thank you!

NickvdMeij commented:
@samdark no you thank you with this awesome generator =D you don't wanna know how much time this has saved us in the past few years. Glad I had the change to help you guys out for a change :) Keep doing the good work, really appreciated!
